File: pythonGemMigration/pipelines/Pipeline_first/code/pipeline_first/graph/source_6.py
from pyspark.sql import *
from pyspark.sql.functions import *
from pyspark.sql.types import *
from prophecy.utils import *
from prophecy.libs import typed_lit
from pipeline_first.config.ConfigStore import *
from pipeline_first.udfs.UDFs import *

def source_6(spark: SparkSession) -> DataFrame:
    import os

    return spark.read\
        .format("jdbc")\
        .option("url", Config.JDBC_URL)\
        .option("user", Config.JDBC_USER_STRING)\
        .option("password", Config.JDBC_PASSWORD_STRING)\
        .option("query", Config.SQL_QUERY)\
        .option("pushDownPredicate", True)\
        .option("driver", Config.DRIVER_NAME)\
        .load()

File: jdbc_gem_scala/pipelines/FirstPipeline/code/src/main/scala/io/prophecy/pipelines/firstpipeline/graph/source6.scala
package io.prophecy.pipelines.firstpipeline.graph

import io.prophecy.libs._
import io.prophecy.pipelines.firstpipeline.config.Context
import org.apache.spark._
import org.apache.spark.sql._
import org.apache.spark.sql.functions._
import org.apache.spark.sql.types._
import org.apache.spark.sql.expressions._
import java.time._

object source6 {

  def apply(context: Context): DataFrame = {
    val Config = context.config
    var reader = context.spark.read.format("jdbc")
    reader = reader
      .option("url",                Config.JDBC_URL)
      .option("user",               sys.env("JDBC_USERNAME"))
      .option("password",           sys.env("JDBC_PASSWORD"))
      .option("pushDownPredicate",  true)
      .option("driver",             "com.mysql.jdbc.Driver")
    reader = reader.option("query", Config.SQL_QUERY)
    var df = reader.load()
    df
  }

}
